runtime/cpp/instance_segmentation/yolov5seg/yolov5seg_example.cpp

```diff
@@ -1,404 +1,318 @@
-        feature->m_buffers.release_write_buffer();
-        if (HAILO_SUCCESS != status) {
-            std::cerr << "Failed reading with status = " <<  status << std::endl;
-            return status;
-        }
+    auto configure_params = hef->create_configure_params(HAILO_STREAM_INTERFACE_PCIE);
+    if (!configure_params) {
+        return make_unexpected(configure_params.status());
     }
 
-    read_time_vec = std::chrono::high_resolution_clock::now();
-    return HAILO_SUCCESS;
-}
+    auto network_groups = device.configure(hef.value(), configure_params.value());
+    if (!network_groups) {
+        return make_unexpected(network_groups.status());
+    }
 
-hailo_status write_all(InputVStream& input_vstream, std::string video_path, 
-                        std::chrono::time_point<std::chrono::system_clock>& write_time_vec, std::vector<cv::Mat>& frames) {
-    m.lock();
-    std::cout << CYAN << "-I- Started write thread: " << info_to_str(input_vstream.get_info()) << std::endl << RESET;
-    m.unlock();
+    if (1 != network_groups->size()) {
+        std::cerr << "Invalid amount of network groups" << std::endl;
+        return make_unexpected(HAILO_INTERNAL_FAILURE);
+    }
 
-    hailo_status status = HAILO_SUCCESS;
-    
-    auto input_shape = input_vstream.get_info().shape;
-    int height = input_shape.height;
-    int width = input_shape.width;
+    return std::move(network_groups->at(0));
+}
 
+template <typename T> hailo_status write_all(std::vector<InputVStream> &input, std::string &video_path,  
+                                            int height, int width, int channels, std::vector<cv::Mat>& frames) {
+    std::cout << "-I- Started write thread " << video_path << std::endl;
     cv::VideoCapture capture(video_path);
+    int i=0; 
+    cv::Mat frame;
     if(!capture.isOpened())
         throw "Unable to read video file";
-    
-    int i = 0;
-    cv::Mat org_frame;
-
-    write_time_vec = std::chrono::high_resolution_clock::now();
-    for(;;) {
-        capture >> org_frame;
-        if(org_frame.empty()) {
+    for( ; ; ) {
+        capture >> frame;
+        if(frame.empty()) {
             break;
-            }
+        }
+        
+        if (frames[i].channels() == 3){
+            cv::cvtColor(frame, frame, cv::COLOR_BGR2RGB);
+        }
 
-        cv::resize(org_frame, frames[i], cv::Size(height, width), 1);
+        if (frame.rows != height || frame.cols != width){
+            cv::resize(frame, frame, cv::Size(width, height), cv::INTER_AREA);
+        }
 
-        input_vstream.write(MemoryView(frames[i].data, input_vstream.get_frame_size())); // Writing height * width, 3 channels of uint8
-        if (HAILO_SUCCESS != status)
-            return status;
-        i++;
-    }
+        frames[i] = frame.clone();
+        auto status = input[0].write(MemoryView(frame.data, height * width * channels)); // Writing height * width, 3 channels of uint8
```

Comment:
The height * width * channels is too restricting and will not support float32 or 16-bit inputs (even through it will never happen in Hailo).
It's better to use the "input_vstream.get_info().get_frame_size()" method to get the correct number of bytes automatically.